src/common/checksum.c

```diff
@@ -494,7 +582,8 @@ daos_csummer_calc(struct daos_csummer *obj, d_sg_list_t *sgl,
 	int		rc = 0;
 	uint32_t	csum_nr = 0;
 
-	if (!(daos_csummer_initialized(obj) && iod && sgl))
+	if (!(daos_csummer_initialized(obj) &&
+	      csum_iod_is_supported(obj->dcs_chunk_size, iod) && sgl))
```

Comment:
should this be !csum_iod_is_supported?

Reply:
No, the whole thing is negated. Maybe it's too confusing that way. I'll change it so the logic is easier to see.